tez-api/src/main/java/org/apache/tez/client/TezClient.java

```diff
@@ -1286,4 +1286,12 @@ public static ApplicationId appIdfromString(String appIdStr) {
               + appIdStr, n);
     }
   }
+
+  public String getAmHost() {
+    return frameworkClient.getAmHost();
+  }
+
+  public int getAmPort() {
+    return frameworkClient.getAmPort();
+  }
```

Comment:
If anyone calls these methods before ``start()`` is invoked in that case ``frameworkClient`` would be `null` right? So we need to check for `null`?

Reply:
right, I'll make this change
a returned null value can mean that frameworkClient is not initialized at all, or initialized but hasn't connected to the AM, from the client's point of view, those are the same